### PR TITLE
Ticket/aotech 6700 duplicate terms

### PR DESCRIPTION
**Pull request by phatsk**
Handles doing full term-data syncs when syncing posts (maintains slug and description). Also uses `get_the_terms` in place of `wp_get_object_terms`, which could prove useful for performance.

### CODE REVIEW

File: includes/class-api.php
Comment (jmichaelward) [inline includes/class-api.php]: Curious about the rationale for the conditional here. Couldn't we just bring over the description regardless of whether the string has any length? It'd bring over the empty string, right?

File: includes/class-api.php
Comment (jmichaelward) [inline includes/class-api.php]: Previously, this method accepted `$terms`, which can be an array|int|string, but we're plucking just the slug out of that. Are there potential regressions that we haven't considered by making this change? I'm not familiar enough with what this block actually _does_ to know the impact, but want to make sure we're being cognizant of the result.

Comment (rakeshdx) [reply includes/class-api.php]: Yeah, Why not insert the whole term array?

Comment (phatsk) [reply includes/class-api.php]: The `wp_list_pluck` here is returning an array of all of the term slugs to be attached to the taxonomy. This doesn't actually insert terms that don't exist, only terms that do exist.

Comment (rakeshdx) [reply includes/class-api.php]: so it's an update than an insert?

**Review by rakeshdx — APPROVED** [review]

File: includes/class-api.php
Comment (jrfoell) [inline includes/class-api.php]: Just want to make sure this is meant to be committed ;)

**Review by jrfoell — CHANGES_REQUESTED** [review]
Just want to confirm that debug should be in here

**Comment by phatsk** [comment]
Working out some issues with @jmichaelward , putting this PR on hold.

**Comment by phatsk** [comment]
Fixed some issues with term meta insertion when sending a post. This is ready for review again.

File: includes/class-api.php
Comment (jmichaelward) [inline includes/class-api.php:1136]: We don't have a method of resolving this using Viacom processes, do we?

File: includes/class-api.php
Comment (jmichaelward) [inline includes/class-api.php]: Let's spell this out `$term_result`.

File: includes/class-api.php
Comment (phatsk) [reply includes/class-api.php:1136]: No, I resolve these manually when I merge to master and tag.

File: includes/class-api.php
Comment (phatsk) [reply includes/class-api.php]: Oh yeah. :derp:

**Review by jmichaelward — APPROVED** [review]
Seems good to me. Please take a look at my two pieces of minor feedback and address them before merging, otherwise, this should be all set.

**Review by rakeshdx — APPROVED** [review]

**Pull request merged**
